plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -256,534 +254,666 @@ struct block_time_tracker {
+   void     schedule_production_loop();
+   void     schedule_maybe_produce_block(bool exhausted);
+   void     produce_block();
+   bool     maybe_produce_block();
+   bool     block_is_exhausted() const;
+   bool     remove_expired_trxs(const fc::time_point& deadline);
+   bool     remove_expired_blacklisted_trxs(const fc::time_point& deadline);
+   bool     process_unapplied_trxs(const fc::time_point& deadline);
+   void     process_scheduled_and_incoming_trxs(const fc::time_point& deadline, unapplied_transaction_queue::iterator& itr);
+   bool     process_incoming_trxs(const fc::time_point& deadline, unapplied_transaction_queue::iterator& itr);
+
+   struct push_result {
+      bool block_exhausted = false;
+      bool trx_exhausted   = false;
+      bool failed          = false;
+   };
+   push_result push_transaction(const fc::time_point&                       block_deadline,
+                                const transaction_metadata_ptr&             trx,
+                                bool                                        api_trx,
+                                bool                                        return_failure_trace,
+                                const next_function<transaction_trace_ptr>& next);
+   push_result handle_push_result(const transaction_metadata_ptr&             trx,
+                                  const next_function<transaction_trace_ptr>& next,
+                                  const fc::time_point&                       start,
+                                  chain::controller&                          chain,
+                                  const transaction_trace_ptr&                trace,
+                                  bool                                        return_failure_trace,
+                                  bool                                        disable_subjective_enforcement,
+                                  account_name                                first_auth,
+                                  int64_t                                     sub_bill,
+                                  uint32_t                                    prev_billed_cpu_time_us);
+   void        log_trx_results(const transaction_metadata_ptr& trx, const transaction_trace_ptr& trace, const fc::time_point& start);
+   void        log_trx_results(const transaction_metadata_ptr& trx, const fc::exception_ptr& except_ptr);
+   void        log_trx_results(const packed_transaction_ptr& trx,
+                               const transaction_trace_ptr&  trace,
+                               const fc::exception_ptr&      except_ptr,
+                               uint32_t                      billed_cpu_us,
+                               const fc::time_point&         start,
+                               bool                          is_transient);
+   void        add_greylist_accounts(const producer_plugin::greylist_params& params) {
```

Comment:
Odd that `add_greylist_accounts` has this indentation while `remove_greylist_accounts` does not. I'm not a fan of more than one space after `void` (return types).

Reply:
Prefer no space between return value and method name, but no big deal.